src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -1006,6 +1006,12 @@ class TagReferenceSegment(ObjectReferenceSegment):
     type = "tag_reference"
 
 
+class StorageReferenceSegment(ObjectReferenceSegment):
+    """A reference to a storage location."""
+
+    type = "storage_reference"
+
+
```

Comment:
Is there a reason this is in ANSI rather than in Snowflake dialect?

Reply:
I put in ansi because all other reference segments are put in `ansi_dialect` and another reason is this form of `storage_reference_segment` might be useful for some other dialects which inherit ansi. But it only be used in snowflake so far